packages/worker/src/queue/queue-consumer.ts

```diff
@@ -107,10 +104,13 @@ export class WorkerQueueConsumer {
 
     this.isProcessing = true;
     const data = job.data;
-    
+
     try {
       logger.info(`Processing thread message job ${job.id} for bot ${data.botId}, thread ${data.threadId}`);
 
+      // Set user context for any database operations
+      process.env.CURRENT_USER_ID = data.userId;
```

Comment:
don't set this at runtime. the worker can have the USER_ID as env schdeduled from the orchestrator.